### ISSUE TITLE
Memory leak while switching the theme

### ISSUE THREAD

**Issue by Mugurell** [duckduckgo/Android#2804]
```
┬───
│ GC Root: Thread object
│
├─ android.net.ConnectivityThread instance
│ Leaking: NO (PathClassLoader↓ is not leaking)
│ Thread name: 'ConnectivityThread'
│ ↓ Thread.contextClassLoader
├─ dalvik.system.PathClassLoader instance
│ Leaking: NO (LocalBroadcastManager↓ is not leaking and A ClassLoader is
│ never leaking)
│ ↓ ClassLoader.runtimeInternalObjects
├─ java.lang.Object[] array
│ Leaking: NO (LocalBroadcastManager↓ is not leaking)
│ ↓ Object[5127]
├─ androidx.localbroadcastmanager.content.LocalBroadcastManager class
│ Leaking: NO (a class is never leaking)
│ ↓ static LocalBroadcastManager.mInstance
│ ~~~~~~~~~
├─ androidx.localbroadcastmanager.content.LocalBroadcastManager instance
│ Leaking: UNKNOWN
│ Retaining 203.0 kB in 3625 objects
│ mAppContext instance of com.duckduckgo.app.global.DuckDuckGoApplication
│ ↓ LocalBroadcastManager.mReceivers
│ ~~~~~~~~~~
├─ java.util.HashMap instance
│ Leaking: UNKNOWN
│ Retaining 248 B in 11 objects
│ ↓ HashMap[key()]
│ ~~~~~~~
├─ com.duckduckgo.mobile.android.ui.
│ ThemingKt$registerForThemeChangeBroadcast$receiver$1 instance
│ Leaking: UNKNOWN
│ Retaining 202.1 kB in 3594 objects
│ Anonymous subclass of android.content.BroadcastReceiver
│ $this_registerForThemeChangeBroadcast instance of com.duckduckgo.mobile.
│ android.themepreview.ui.AppComponentsActivity with mDestroyed = true
│ ↓ ThemingKt$registerForThemeChangeBroadcast$receiver$1.
│ $this_registerForThemeChangeBroadcast
│ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
╰→ com.duckduckgo.mobile.android.themepreview.ui.AppComponentsActivity instance
​ Leaking: YES (ObjectWatcher was watching this because com.duckduckgo.
​ mobile.android.themepreview.ui.AppComponentsActivity received
​ Activity#onDestroy() callback and Activity#mDestroyed is true)
​ Retaining 202.0 kB in 3593 objects
​ key = 6f911e48-897d-40cd-9868-2fa21afdcfce
​ watchDurationMillis = 8718
​ retainedDurationMillis = 3716
​ mApplication instance of com.duckduckgo.app.global.DuckDuckGoApplication
​ mBase instance of androidx.appcompat.view.ContextThemeWrapper
```

Happened only once.

**Comment by cmonfortep** [comment]
Thanks, that's an internal screen and we are aware of it, but low priority.
